codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -587,7 +591,7 @@ private State readHeaders(ByteBuf buffer) {
 
         State nextState;
 
-        if (isContentAlwaysEmpty(message)) {
+        if (isContentAlwaysEmpty(message) || HttpUtil.hasEmptyContent(message)) {
```

Comment:
@jza70 ah sorry I missed that it calls `HttpUtil.setTransferEncodingChunked(message, false);` (I missed the false) here which means the Content-Length is still preserved correctly, right ?

In this case I think you are correct and all is good.